sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchClientTestBase.java

```diff
@@ -56,59 +70,62 @@ protected void beforeTest() {
     }
 
     MapsSearchClientBuilder getMapsSearchAsyncClientBuilder(HttpClient httpClient,
-            MapsSearchServiceVersion serviceVersion) {
-        // System.out.println("beginning httpclient " + httpClient);
+                                                            MapsSearchServiceVersion serviceVersion) {
         MapsSearchClientBuilder builder = new MapsSearchClientBuilder()
-                // .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() :
-                // httpClient)
-                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
-                .serviceVersion(serviceVersion);
-        String endpoint = getEndpoint();
-        if (getEndpoint() != null) {
-            builder.endpoint(endpoint);
-        }
-        // System.out.println("after httpclient " + httpClient);
-        if (getTestMode() == TestMode.RECORD) {
-            builder.addPolicy(interceptorManager.getRecordPolicy());
-        }
-        if (getTestMode() == TestMode.PLAYBACK) {
-            builder.credential(new AzureKeyCredential(FAKE_API_KEY)).httpClient(interceptorManager.getPlaybackClient());
-        } else {
-            builder.credential((new AzureKeyCredential(
-                    Configuration.getGlobalConfiguration().get("SUBSCRIPTION_KEY"))));
+            .pipeline(getHttpPipeline(httpClient))
+            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+            .serviceVersion(serviceVersion);
+
+        if (interceptorManager.isPlaybackMode()) {
+            builder.endpoint("https://localhost:8080");
         }
+
         return builder;
     }
 
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode() || interceptorManager.isPlaybackMode()) {
+            interceptorManager.addSanitizers(
+                Collections.singletonList(
+                    new TestProxySanitizer("subscription-key", ".+", "REDACTED", TestProxySanitizerType.HEADER)));
```

Comment:
nit: why not just use the core-test unrelased dependency?

Reply:
I will update all dependencies to use the new version once it comes out.